cypress/support/pageObjects/OrangeHRM/addEmployee.ts

```diff
@@ -63,27 +85,98 @@ class addEmployee {
     });
   }
 
-  createLoginDetails(username: string, password: string) {
-    const currentDate = new Date();
+  createLoginDetails(username: string, password: string): Cypress.Chainable<any> {
+    return cy.wrap(undefined).then(() => {
+      cy.request({
+        method: "POST",
+        url: "/web/index.php/api/v2/admin/users",
+        body: {
+          empNumber: this.getEmpNumber(),
+          password: password,
+          status: true,
+          userRoleId: 2,
+          username: username
+        },
+      }).then((response) => {
+        console.log(response.body.data.userName);
+        this.setUserName(response.body.data.userName);
+        expect(response).property("status").to.equal(200);
+      });
+    });
+  }
+
+  logout() {
+    this.elements.dropDown().click()
+    this.elements.logout().click()
+  }
+
+  addLeaveEntitlement() {
     cy.request({
       method: "POST",
-      url: "/web/index.php/api/v2/admin/users",
-      body: {
-        empNumber: this.getEmpNumber(),
-        password: password,
-        status: true,
-        userRoleId: 2,
-        username: username
-      },
+      url: `/web/index.php/api/v2/leave/leave-entitlements`,
+      body:{
+        empNumber: this.getEmpNumber(), 
+        leaveTypeId: 8, 
+        fromDate: "2023-01-01", 
+        toDate: "2023-12-31", 
+        entitlement: "40"
+      }
+      
+    }).then((response) => {
+      expect(response).property("status").to.equal(200);
+    });
+  }
+  
+  applyLeave(){
+    cy.request({
+      method: "POST",
+      url: `/web/index.php/api/v2/leave/leave-requests`,
+      body:{
+        leaveTypeId: 8, 
+        fromDate: "2023-10-30", 
+        toDate: "2023-10-30", 
+        comment: null
+      }
+      
     }).then((response) => {
+      console.log(response.body.data.id);
+      this.setLeaveId(response.body.data.id);
       expect(response).property("status").to.equal(200);
     });
   }
 
-  logout(){
-    this.elements.dropDown().click()
-    this.elements.logout().click()
+  adminApprovesLeave(){
+    cy.request({
+      method: "PUT",
+      url: `/web/index.php/api/v2/leave/employees/leave-requests/${this.getLeaveId()}`,
+      body:{
+        action: "APPROVE"
+      }
+      
+    }).then((response) => {
+      console.log(response);
+      expect(response).property("status").to.equal(200);
```

Comment:
No need for this assertion cuz we use the API here not make API testing to check if the API is success or not